module/zfs/zfs_ioctl.c

```diff
@@ -1442,9 +1442,7 @@ zfsvfs_rele(zfsvfs_t *zfsvfs, void *tag)
 {
 	rrm_exit(&zfsvfs->z_teardown_lock, tag);
 
-	if (zfsvfs->z_sb) {
-		deactivate_super(zfsvfs->z_sb);
```

Comment:
Functionally this works, but I don't think we want the interface to  allow `zfs_vfs_rele()` to fail, it would be preferable for the return type to be `void`.  Why don't we introduce an additional `boolean_t zfs_vfs_held(zfsvfs_t *)` wrapper, so it would be something like this:

```c
if (zfs_vfs_held(zfsvfs)) {
        zfs_vfs_rele(zfsvfs);
} else {
        ...
```

and,

```c
boolean_t
zfs_vfs_held(zfsvfs_t *zfsvfs)
{
        return (zfsvfs->z_sb != NULL);
}

void
zfs_vfs_rele(zfsvfs_t *zfsvfs)
{
        deactivate_super(zfsvfs->z_sb);
}
```

This has the additional advantage that we're not ignoring the return type for the other `zfs_vfs_rele()` calls.